addons/mrp_subcontracting/models/stock_warehouse.py

```diff
@@ -26,6 +26,17 @@ class StockWarehouse(models.Model):
         'stock.picking.type', 'Subcontracting Resupply Operation Type',
         domain=[('code', '=', 'outgoing')])
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super().create(vals_list)
+        self._update_subcontracting_locations_rules()
+        return res
+
+    def write(self, vals):
+        res = super().write(vals)
+        self._update_subcontracting_locations_rules()
```

Comment:
should we really call `_update_subcontracting_locations_rules` every time something in changed in a warehouse?